dat.js

```diff
@@ -0,0 +1,117 @@
+var assert = require('assert')
+var path = require('path')
+var multicb = require('multicb')
+var importFiles = require('./lib/import-files')
+var createNetwork = require('./lib/network')
+var stats = require('./lib/stats')
+
+module.exports = Dat
+
+function Dat (archive, db, opts) {
+  if (!(this instanceof Dat)) return new Dat(archive, db, opts)
+  if (typeof opts === 'undefined') return Dat(archive, null, db)
+  assert.ok(archive, 'archive required')
+  // assert.ok(db, 'database required') // maybe not be required for multidrive...
+  assert.ok(opts.dir, 'opts.directory required')
+
+  this.path = path.resolve(opts.dir)
+  this.options = opts
+
+  this.archive = archive
+  this.db = db
+  this.key = archive.key // only resumed/owned archives will have keys here
+  this.live = archive.live
+  this.owner = archive.owner
+  this.resumed = archive.resumed
+}
+
+Dat.prototype.join =
+Dat.prototype.joinNetwork = function (opts) {
+  if (this.network) return this.network.join(this.archive.discoveryKey)
+  var self = this
+
+  var network = self.network = createNetwork(self.archive, opts)
+  self.options.network = network.options
+
+  network.swarm = network // 1.0 backwards compat
+  if (self.owner) return network
+
+  network.once('connection', function () {
+    // automatically open archive and set exposed values
+    self.archive.open(function () {
+      // self.owner = self.archive.owner // For future multi-writer?
+      self.live = self.archive.live
+    })
+  })
+  return network
+}
+
+Dat.prototype.leave =
+Dat.prototype.leaveNetwork = function () {
+  if (!this.network) return
+  this.network.leave(this.archive.discoveryKey)
+}
+
+Dat.prototype.trackStats = function (opts) {
+  opts = opts || {}
+  assert.ok(opts.db || this.db, 'Dat needs database to track stats')
+  this.stats = stats(this.archive, opts.db || this.db)
+  return this.stats
+}
+
+Dat.prototype.importFiles = function (target, opts, cb) {
+  if (typeof target !== 'string') return this.importFiles('', target, opts)
+  if (typeof opts === 'function') return this.importFiles(target, {}, opts)
+  if (!this.archive.owner) return cb(new Error('Must be archive owner to import files.'))
```

Comment:
Ya. It'll be true if the archive is local. For future multi-writer, we can set it after a [downloaded archive](https://github.com/datproject/dat-node/blob/multidrive-support/dat.js#L42) is opened.